language-support/hs/bindings/examples/nim/src/DA/Ledger/App/Nim/NimLedger.hs

```diff
@@ -31,9 +33,22 @@ run timeout ls  = runLedgerService ls timeout (configOfPort port)
 connect :: Logger -> IO Handle
 connect log = do
     lid <- run 5 getLedgerIdentity
-    ids <- run 5 $ listPackages lid
-    [_,pid,_] <- return ids -- guess which is the Nim package -- TODO: fix this properly!
-    return Handle{log,lid,pid}
+
+    discovery <- run 5 $ do
+        pids <- listPackages lid
+        forM pids $ \pid -> do
+            getPackage lid pid >>= \case
+                Nothing -> return (pid, False)
+                Just package -> do
+                    return (pid, containsNim package)
+
+    case List.filter snd discovery of
+        [] -> fail "cant find package containing Nim"
+        xs@(_:_:_) -> fail $ "found multiple packages containing Nim: " <> show (map fst xs)
+        [(pid,_)] -> return Handle{log,lid,pid}
+
+containsNim :: LF.Package -> Bool
+containsNim package = "Nim" `isInfixOf` show package -- TODO: be more principled
```

Comment:
Can we at least check some module name or something a bit more specific than just grepping through the `show` output of the whole package? I also don’t really understand why we cannot read the dar here like we do in the tests and do this properly.